vendor-extra/ContentPageBundle/src/Controller/ContentController.php

```diff
@@ -33,18 +51,37 @@ public function __invoke(string $id) : Response
                 ]
             )
             ->then(
-                function (ResponseInterface $response) {
+                function (ResponseInterface $response) use ($request) {
                     $dom = FluentDOM::load((string) $response->getBody());
                     $dom->registerNamespace('libero', 'http://libero.pub');
 
-                    /** @var string $title */
-                    $title = $dom('string(/libero:item/libero:front/libero:title)');
+                    /** @var DOMNodeList|Element[] $frontList */
+                    $frontList = $dom('/libero:item/libero:front[1]');
+                    $front = $frontList->item(0);
 
-                    if ('' === $title) {
-                        throw new UnexpectedValueException('Could not find a title');
+                    if (!$front instanceof Element) {
+                        throw new UnexpectedValueException('Could not find a front');
                     }
 
-                    return new Response("<html><body>${title}</body></html>");
+                    $context = [
+                        'lang' => $request->getLocale(),
+                        'dir' => 'right-to-left' === Misc::getCharacterOrder($request->getLocale()) ? 'rtl' : 'ltr',
+                    ];
+
+                    $header = $this->converter->convert($front, '@LiberoPatterns/content-header.html.twig', $context);
```

Comment:
sounds like `$this->converter`, `$this->twig` and `$this->template` are tied together but it's early to say how to abstract them into an object until we see more examples

Reply:
`$this->twig` and `$this->template` are definitely tied together.

Reply:
Leave for now. Might need a little trickery (maybe a template name alias?), but let's see how it plays out first (might be ok to have the same base for all instances, or customisable).